src/core/plugin.js

```diff
@@ -0,0 +1,59 @@
+const {ChildProcess} = require("child_process");
+const treekill = require("treekill");
+const {execute, checkServer} = require("../core/executor");
+const Locale = require("../locales")("service.run");
+
+module.exports = function ({command, config}, processor = {}) {
+  if (!processor.BeforeAll || !processor.AfterAll) {
+    throw new Error(
+      "Please provide a processor containing the methods:  BeforeAll, AfterAll"
+    );
+  }
+
+  processor.Before(async function (scenario) {
+    // In order to fetch the dynamic data from extenral sources.
+    // Let's parse the scenario to get all the placeholder
+    await this.data.parse(scenario);
```

Comment:
Don't really understand this. Could you explain a bit more? 🙈

Reply:
Will do offline